File: CPP/Integer_To_Roman.cpp
// https://leetcode.com/problems/integer-to-roman/

class Solution {
public:
    string intToRoman(int num) {
        vector<string> sym = {"I","IV","V","IX","X","XL","L","XC","C","CD","D","CM","M"};
        vector<int> numbers = {1,4,5,9,10,40,50,90,100,400,500,900,1000};
        string ans="";
        int length=12;
        while(num>0) {
            int div=num/numbers[length];
            num = num%numbers[length];
            while(div--) 
                ans.append(sym[length]);
            length--;
        }
        return ans;
    }
};

File: Python/integer-to-roman.py
class Solution:
    def intToRoman(self, n: int) -> str:
        symbols = [["I", 1], ["IV", 4], ["V", 5], ["IX", 9], ["X", 10], ["XL", 40], ["L", 50], ["XC",90], ["C", 100], ["CD", 400], ["D", 500], ["CM", 900],["M", 1000]]
        final = ""
        for symbols, v in reversed (symbols):
            if n // v:
                count = n // v
                final += (symbols* count)
                n = n % v
        return final
